### PR TITLE
Remove the unknown category

### PR DESCRIPTION
**Pull request by oltionchampari**
Closes Caleydo/tdp_publicdb#99

### Summary

* Commented out the category `unknown` that represents the missing values.

### Documentation
We assign the value `'null'`(type string) to the missing rows. For the purpose of this ticket it is not necessary removing the `missingCategory` property in the config.ts in tdp_publicdb.
https://github.com/Caleydo/tdp_publicdb/blob/d8b3eceb66391f49dded9753434b79e64a610d2c/src/config.ts#L341-L349

#### Regarding the compatibility of older session: 

The issue is that the the unknown values will not be filtered anymore since the category unknown does not exist anymore.
The filter can be one of the 3 below combinations:
* `['null'] ` --> filter everything but the unknown values
* `['true', 'null']`--> filter out values that are false(NonMutated)
* `['true", 'false']` --> filter out missing values

For each case we would like to set the correct filter when an older graph gets imported and to achieve that we would have to translate the filters to work with this implementation.

For the first two filters it would be possible to set the correct filter but for the third the filter would be hard to translate since the unknown category is missing and the filter filters out that category.

unkown_category

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -11,15 +11,15 @@ export const copyNumberCat = [
   {value: 0, name: 'NORMAL', color: '#dcdcdc', border: 'transparent'},
   //{value: 1, name: 'Low level amplification', color: '#f4a582'},
   //{value: 2, name: 'High level amplification', color: '#ca0020'},
-  {value: 'null', name: 'Unknown', color: '#FCFCFC', border: '#dcdcdc'}
+  // {value: 'null', name: 'Unknown', color: '#FCFCFC', border: '#dcdcdc'}
```
Comment (thinkh) [inline src/constants.ts:14]: We cannot remove this category as we are using them also in the OncoPrints.

https://github.com/Caleydo/tdp_gene/blob/2ff1964b81f4651fc7367c6564b13b7ae967e644/src/views/AOncoPrint.ts#L238-L240

File: src/constants.ts [hunk 2]
```diff
@@ -11,15 +11,15 @@ export const copyNumberCat = [
...
 ];
-export const unknownCopyNumberValue: any = copyNumberCat[copyNumberCat.length-1].value;
+export const unknownCopyNumberValue: any = 'null';
 
 export const mutationCat = [
   {value: 'true', name: 'Mutated', color: '#1BA64E', border: 'transparent'},
   {value: 'false', name: 'Non Mutated', color: '#aaa', border: 'transparent'},
-  {value: 'null', name: 'Unknown', color: 'transparent', border: '#999'}
+  // {value: 'null', name: 'Unknown', color: 'transparent', border: '#999'}
```
Comment (thinkh) [inline src/constants.ts:21]: We cannot remove this category as we are using them also in the OncoPrints.

https://github.com/Caleydo/tdp_gene/blob/2ff1964b81f4651fc7367c6564b13b7ae967e644/src/views/AOncoPrint.ts#L245-L247

File: src/constants.ts [hunk 3]
```diff
@@ -11,15 +11,15 @@ export const copyNumberCat = [
   {value: 0, name: 'NORMAL', color: '#dcdcdc', border: 'transparent'},
   //{value: 1, name: 'Low level amplification', color: '#f4a582'},
   //{value: 2, name: 'High level amplification', color: '#ca0020'},
-  {value: 'null', name: 'Unknown', color: '#FCFCFC', border: '#dcdcdc'}
+  //{value: 'null', name: 'Unknown', color: '#FCFCFC', border: '#dcdcdc'}
 ];
-export const unknownCopyNumberValue: any = copyNumberCat[copyNumberCat.length-1].value;
+export const unknownCopyNumberValue: any = NaN;
```
Comment (oltionchampari) [inline src/constants.ts:16]: Changed the value of the `unknownCopyNumberValue` from `'null'` to `NaN` to make it compatible with the new tourdino filter. The function `isMissingValue()` from LineUp does not filter out `'null'` values. An alternative option would be to adapt the `isMissingValue()` function.

Comment (thinkh) [reply src/constants.ts:16]: I assume you are referring to PR https://github.com/Caleydo/tourdino/pull/55.

Comment (oltionchampari) [reply src/constants.ts:16]: Yes

**Comment by thinkh** [comment]
@oltionchampari Thanks for your changes and testing. The OncoPrint looks fine.

While testing I found a bug. I'm not sure, if this bug is a general issue or releated to this PR. Please test if you can confirm this behavior.

### Steps to reproduce

1. Add a categorical score column (e.g., _AA Mutated_)
2. Check Filter missing values
3. Uncheck filter missing values


### Observed behavior

ordino-reset-categorical-score-column

The icon is still highlighted and only resetting the filter removes the highlight.

### Expected behavior

Unchecking the missing values should remove the highlight (if no other filter is set). For numerical score columns it works correctly.

ordino-reset-numerical-score-column

**Comment by thinkh** [comment]
I just checked this behavior on https://ordino-daily.caleydoapp.org, but with the _Unknown_ category, with the same result. Hence, it must be a general problem with LineUp v3, categorical score columns and maybe the provenance graph.

ordino-reset-categorical-score-filter

@oltionchampari Could you please check whether wrong behavior is still present with LineUp v4 (and all the provenance graph fixes we made)?

**Comment by oltionchampari** [comment]
In `LineUp v4` the above described bug for the categorical column has been fixed.

categorical_filter_lineup4

The bug is however present for the string column. It occurs only when checking the checkbox in the Sidepanel. Filtering missing from the filter dialog behaves correctly. This is most probably related to the [live preview](https://github.com/datavisyn/tdp_core/pull/358) ticket.

string_filter_lineup4

**Comment by thinkh** [comment]
@oltionchampari Thanks for testing. I opened a new issue for the String column bug in https://github.com/datavisyn/tdp_core/issues/374. I will merge this PR as it will be solved with LineUp v4.

**Pull request merged**